### PR TITLE
Externalize Principal component

### LINKED ISSUE #151: Unit test: Injection not found
For some component tests, we have some warnings like 

```
 console.error node_modules/vue/dist/vue.runtime.common.js:589
      [Vue warn]: Injection "translationService" not found
      
      found in
      
      ---> <ChangePassword>
             <Root>
```

That's because we use `Principal` component as mixin, which to have the `translationService` injected. We have to find a way to inject it when mounting these components in unit test.

### PR DESCRIPTION
**Pull request by hdurix**
Fix #151

### CODE REVIEW

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { mixins } from 'vue-class-component';
-import { Component, Inject } from 'vue-property-decorator';
+import { Component, Inject, Vue } from 'vue-property-decorator';
 import UserManagementService from './user-management.service';
-import Principal from '../../account/principal';
+import Principal from '@/account/principal';
 
 @Component
-export default class <%=jhiPrefixCapitalized%>UserManagementComponent extends mixins(Principal) {
+export default class JhiUserManagementComponent extends Vue {
```
Comment (sahbi-ktifa) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs]: Why did you remove the templated prefix?

Comment (hdurix) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs]: I didn't do it on purpose! Thanks :)

**Comment by atomfrede** [comment]
@hdurix Are you still working on it or should we merge it?

**Comment by pascalgrimaud** [comment]
I suggest to merge it as it impacts a lot of files.
Then it can be improve later in another PR if it was not finished.
Are you OK with that @atomfrede ?

**Comment by atomfrede** [comment]
@pascalgrimaud Same for me, so I'll merge it.

**Comment by hdurix** [comment]
It was all done :) Thanks!
